starboard/shared/starboard/player/filter/filter_based_player_worker_handler.cc

```diff
@@ -351,7 +351,7 @@ bool FilterBasedPlayerWorkerHandler::SetPause(bool pause) {
   } else {
     media_time_provider_->Play();
   }
-
+  Update();
```

Comment:
Update() will Schedule() another Update() call internally, which can lead to redundant calls.

Let's remove any scheduled Update() calls inside Update() to ensure that it won't be scheduled multiple times.